cppapi/server/zmqeventsupplier.cpp

```diff
@@ -63,6 +63,38 @@ ZmqEventSupplier *ZmqEventSupplier::_instance = NULL;
 /*		       															*/
 /************************************************************************/
 
+ZmqEventSupplier::Future::Future():is_ready_(false),mutex(),cond(&mutex)
+{
+
+}
+void ZmqEventSupplier::Future::set()
+{
+    omni_mutex_lock lk(mutex);
+    is_ready_ = true;
+    cond.signal();
+    
+}
+
+void ZmqEventSupplier::Future::wait()
+{
+    if(!is_ready_)
+    {
+        omni_mutex_lock lk(mutex);
+        cond.wait();
```

Comment:
Spurious wake-ups may happen and wait() can return without condition variable being notified by other thread (omnithread does not guard us against this). Always perform such waits in a loop and re-evaluate the condition after wake-up. This function should rather look like:
```cpp
mutex.lock();
while (not is_ready_)
{
    cond.wait();
}
mutex.unlock();
```